PlasmaAnchor/anchor.md

```diff
@@ -0,0 +1,117 @@
+## Anchor: Finality Gadget on Tendermint
+
+### Motivation
+In order for Plasma to work, the sidechain must be aware of the rootchain smart contract state. The sidechain must know what has been deposited so that they can be spent, and must also know what positions have exited in order to prevent them from being spent on the sidechain. The main difficulty in acheiving this is the fact that Ethereum has probabilistic finality while Tendermint has instant finality. Thus a Tendermint chain communicating with a PoW chain may commit something on the basis of unfinalized state in the PoW chain.
+
+Having a single node act as an oracle for the sidechain would cause a central point of failure. Including the anchor mechanism in the sidechain consensus state introduces a lot of complexity and is not very modular. Anchor is a seperate process that will run Tendermint consensus to determine the latest rootchain state that a quorum of validators deem finalized. This will be used to communicate to the sidechain the latest finalized rootchain state to process deposits and exits.
+
+The sidechain must track `Deposits` in the rootchain so that users can spend finalized deposits on the sidechain. The sidechain should also track `Exits` in order to delete positions that are no longer spendable (space-saving) and prevent users from spamming the chain by spending exited positions.
+
+### App State of Anchor
+The app state of anchor contains the following data in substores:
+
+Rootchain Blocks: `Map{Number -> BlockHash}`
+
+Deposits: `Map{Number -> Deposit}`
+
+where `Deposit` is:
+
+```go
+type Deposit struct {
+    Owner  ethcmn.Address
+    Amount uint64
+}
+```
+
+and Exits: `Maps{Position -> Exit}`
+
+where `Exit` is:
+
+```go
+type Exit struct {
+    Owner    ethcm.Address
+    Position [4]uint64
+}
+```
+
+**Note:** RootChain Blocks and Deposits are continuous from 1 to the current objectively finalized value, thus both are simply lists represented in an sdk store.
+
+Let the current objectively finalized block be `B` and the currently objectively finalized deposit be `N`.
+
+### Msgs
+
+Msgs in anchor are submitted by each sidechain validator once per block
+
+```go
+// Implement sdk.Msg interface
+type AnchorMsg struct {
+    // Msg contains all subjectively finalized data 
+    // since last objectively finalized data
+    RootchainBlockHash [][]byte
+    Deposits           []Deposit
```

Comment:
Why must deposits/exits be passed around as messages in the network?

If the blockhash is already known, can't this information be retrieved locally from the geth node? I was thinking each client would temporarily collect deposits/exits locally per block